apps/trading-e2e/src/support/step_definitions/home-page.step.ts

```diff
@@ -1,7 +1,126 @@
+});
+
+Then(
+  'the server contains at least {int} open markets',
+  function (expectNumber) {
+    expect(this.openMarketData.length).to.be.at.least(expectNumber);
+  }
+);
+
+Then(
+  'I expect the market overlay table to contain at least {int} rows',
+  function (expectNumber) {
+    cy.get('table tr').then((row) => {
+      expect(row.length).to.be.at.least(expectNumber);
+    });
+  }
+);
+
+Then(
+  'each market shown in overlay table exists as open market on server',
+  function () {
+    homePage.getOpenMarketCodes(this.openMarketData).then((openMarketCodes) => {
+      homePage.validateTableCodesExistOnServer(openMarketCodes);
+    });
+  }
+);
+
+Then(
+  'each market shown in overlay table contains content under the last price and change fields',
+  function () {
+    homePage.getOpenMarketCodes(this.openMarketData).then((openMarketCodes) => {
+      homePage.validateTableContainsLastPriceAndChange(openMarketCodes);
+    });
+  }
+);
+
+Then(
+  'each market shown in the full list exists as open market on server',
+  function () {
+    homePage.getOpenMarketCodes(this.openMarketData).each((openMarketCode) => {
+      cy.contains(openMarketCode).should('be.visible');
+    });
+  }
+);
+
+Then(
+  'the oldest market trading in continous mode shown at top of overlay table',
+  function () {
+    homePage
+      .getOldestTradableInstrument(this.openMarketData)
+      .then((oldestTradableInstrument) => {
+        homePage.validateStringIsDisplayedAtTopOfTable(
+          oldestTradableInstrument.instrument.code
+        );
+      });
+  }
+);
+
+Then(
+  'the oldest current trading market is loaded on the trading tab',
+  function () {
+    homePage
+      .getOldestTradableInstrument(this.openMarketData)
+      .then((oldestTradableInstrument) => {
+        cy.getByTestId('market', { timeout: 12000 }).within(() => {
+          cy.get('button')
+            .contains(oldestTradableInstrument.instrument.name)
+            .should('be.visible');
+        });
+      });
+  }
+);
+
+Then(
+  'the most recent current trading market is loaded on the trading tab',
+  function () {
+    homePage
+      .getMostRecentTradableInstrument(this.openMarketData)
+      .then((mostRecentMarketInstrument) => {
+        cy.getByTestId('market', { timeout: 12000 }).within(() => {
+          cy.get('button')
+            .contains(mostRecentMarketInstrument.instrument.name)
+            .should('be.visible');
+        });
+      });
+  }
+);
+
+When('I click the most recent trading market', function () {
+  homePage
+    .getMostRecentTradableInstrument(this.openMarketData)
+    .then((mostRecentMarketInstrument) => {
+      cy.contains(mostRecentMarketInstrument.instrument.code).click();
+    });
+});
+
+When('I click the view full market list', () => {
+  cy.contains('Or view full market list').click();
+  cy.contains('Loading...').should('be.visible');
+  cy.contains('Loading...').should('not.exist');
+});
```

Comment:
I'm seeing quite a few type errors in this file and probably what's failing currently failing the build

Reply:
All typescript stuff now complete